### PR TITLE
Polish car price tutorial 

### PR DESCRIPTION
**Pull request by Yancey0623**
Polish car price tutorial.

### CODE REVIEW

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
+how these features affect the car price.
 
-## The Dataset
+## Car Price Dataset
 
-This tutorial would use the [Car Features and MSRP](https://www.kaggle.com/CooperUnion/cardataset) as the demonstration dataset from kaggle.
+This tutorial would use the [Cars Dataset](https://www.kaggle.com/CooperUnion/cardataset) as the demonstration dataset from [kaggle](https://www.kaggle.com/).
 
-We have used feature engineering to preprocess the raw data and create new features. The feature engineering file is `carprice_preprocessing.py` in the current directory. Finally, We separated the dataset into the train/test dataset and got the `trainSet.csv` and `testSet.csv` files, which is used to train/predict our model. 
+The preprocessed data has already loaded in MySQL, and you can check the Python script [carprie_preprocessing.py](/doc/tutorial/didi/carprice_xgboost/carprie_preprocessing.py) to reproduce the preprocessing.
```
Comment (weiguoz) [inline doc/tutorial/didi/carprice_xgboost/README.md]: "has already loaded in MySQL" => "has been loaded to MySQL" ?

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 2]
```diff
@@ -1,32 +1,35 @@
-# XGBoost on SQLFlow Tutorial
+# Predict the Car Price using XGBoost in SQLFlow
+
+This is a tutorial on how to train/predict an XGBoost model to fit the [Cars Dataset](https://www.kaggle.com/CooperUnion/cardataset). You can find more SQLFlow usage from the [User Guide](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/user_guide.md). In this tutorial, you will learn how to:
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: This tutorial describes how to train an XGBoost model using the Cars Dataset, then how to predict the car price using SQLFlow.

Comment (typhoonzero) [reply doc/tutorial/didi/carprice_xgboost/README.md]: You can find more SQLFlow usage from the => Checkout the User Guide if you are new to SQLFlow.

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 3]
```diff
@@ -1,32 +1,35 @@
...
 
-This is a tutorial on a train/predict XGBoost model in SQLFLow, you can find more SQLFlow usage from the [User Guide](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/user_guide.md). In this tutorial, we will introduce the following:
 - Train an XGBoost model to fit the cars dataset.
-- predict the price of a car using a well-trained model.
-- Use the [SHAP EXPLAINER](https://github.com/slundberg/shap) tool to interpret the output of the trained model to known how these features affect the price.
+- Predict the car price using a well-trained model.
+- Using the [SHAP EXPLAINER](https://github.com/slundberg/shap) toolkit to interpret the trained model that you can know
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: Using => Use

Comment (typhoonzero) [reply doc/tutorial/didi/carprice_xgboost/README.md]: that -> then

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
+how these features affect the car price.
 
-## The Dataset
+## Car Price Dataset
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: The Car Price Dataset

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 5]
```diff
@@ -1,32 +1,35 @@
...
 
-This tutorial would use the [Car Features and MSRP](https://www.kaggle.com/CooperUnion/cardataset) as the demonstration dataset from kaggle.
+This tutorial would use the [Cars Dataset](https://www.kaggle.com/CooperUnion/cardataset) as the demonstration dataset from [kaggle](https://www.kaggle.com/).
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: This tutorial would use -> we are using

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 6]
```diff
@@ -1,32 +1,35 @@
...
 
-We have used feature engineering to preprocess the raw data and create new features. The feature engineering file is `carprice_preprocessing.py` in the current directory. Finally, We separated the dataset into the train/test dataset and got the `trainSet.csv` and `testSet.csv` files, which is used to train/predict our model. 
+The preprocessed data has already loaded in MySQL, and you can check the Python script [carprie_preprocessing.py](/doc/tutorial/didi/carprice_xgboost/carprie_preprocessing.py) to reproduce the preprocessing.
 
-The dataset after preprocessed contains eighty-two features and one label. The features identify the characteristics of a car. Each feature is stored as a single float number. The label `msrp` indicates the car's price. 
+The preprocessed data contains eighty-two features and one label. The features identify the characteristics of a car. Each feature is stored as a single float number. The label `msrp` indicates the car's price.
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: eighty-two -> 82?

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 7]
```diff
@@ -1,32 +1,35 @@
...
 
-The meaning of parts of columns in `trainSet.csv` and `testSet.csv` is shown as follows:
+The meanings of a part of the columns are shown below:
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: Here are some of the column descriptions of the dataset:

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 8]
```diff
@@ -44,9 +47,11 @@ from carprice.train
...
 ```
 
-# Train: Fit Cars Dataset
-Let's train an xgboost model on the dataset. We prefer to train the model for `300 rounds`,
+## Train an XGBoost model using SQLFlow
+
+Let's train an XGBoost model on the dataset. We prefer to train the model for `300 rounds`,
 and using `squarederror` loss function that the SQLFLow extended SQL can be like:
+
 ``` text
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md:55]: why not use sql?

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 9]
```diff
@@ -56,26 +61,26 @@ WITH
 
 `xgboost.gbtree` is the estimator name, `gbtree` is one of the XGBoost booster, you can find more information from [here](https://xgboost.readthedocs.io/en/latest/parameter.html#general-parameters).
 
-To specify the training data, we use standard SQL statements to fetch the traning data from table `carprice.train`:
+To specify the training data, we use standard SQL statements to fetch the training data from table `carprice.train`:
 
 ``` text
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md:66]: why not use sql?

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 10]
```diff
@@ -56,26 +61,26 @@ WITH
...
 SELECT * FROM carprice.train
 ```
 
-We can explicit specify which column is used for features in `COLUMN clause` and which column is used for the label by `LABEL` keyword:
+We can explicitly specify which column is used for features in `COLUMN clause` and which column is used for the label by `LABEL` keyword:
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: is used for features -> is used as features

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 11]
```diff
@@ -56,26 +61,26 @@ WITH
...
+COLUMN engine_hp, engine_cylinders, number_of_doors, highway_mpg, city_mpg, ...
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md:73]: So where to find the full statement when user want to run the SQL in notebook?

**Review by Echo9573 — COMMENTED** [review]
Maybe we can add a SHAP explaining figure to carprice_xgboost/README.md.
Or, whether we can provide .ipythb documentation at the same time?

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 12]
```diff
@@ -91,7 +96,8 @@ LABEL msrp
 INTO sqlflow_models.my_xgb_regression_model;
 ```
 
-# Predict the car's price
+## Predict the Car Price
+
 After training the regression model, let's predict the car price using the trained model.
 
 First, we can specify the trained model by USING clause:
```
Comment (Echo9573) [inline doc/tutorial/didi/carprice_xgboost/README.md:103]: And using a standar SQL to fetch the prediction data:   standar-->standard; 
Than, we can specify the prediction result table by PREDICT clause:
 : than-->then

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 9 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md:66]: The toolkit `markdown-to-ipynb` would convert ` ```sql ` block to code-block in ipynb.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 1 above)
Comment (Echo9573) [inline doc/tutorial/didi/carprice_xgboost/README.md]: carprie_preprocessing --> carprice_preprocessing

**Review by Echo9573 — COMMENTED** [review]
Do we need to download the dataset from kaggle and put it under the test data folder? This may be more convenient for users.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 9 above)
Comment (weiguoz) [reply doc/tutorial/didi/carprice_xgboost/README.md:66]: Appending a white space between ``` and `sql` is a workaround way.
But I don't think it is a good enough solution.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 1 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: done.

**Comment by Yancey0623** [comment]
> Do we need to download the dataset from kaggle and put it under the test data folder? This may be more convenient for users.

@Echo9573  May not, user can download it from the url in the tutorial.

> Maybe we can add a SHAP explaining figure to carprice_xgboost/README.md
Done.

> Or, whether we can provide .ipythb documentation at the same time?

The toolkit `convert_markdown_to_ipynb` would convert the markdown tutorial into ipynb file while building the Docker image, ref:  https://github.com/sql-machine-learning/sqlflow/pull/923/files#diff-ada9de7b49a8cb1f88f4e95950c476ceR10.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 1 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 2 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 4 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: Done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 5 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: Done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 6 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 7 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 8 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md:55]: `convert_markdown_to_ipynb` tookit will convert the ` ```sql ` block into an executable code-block in ipynb.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 10 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md]: done.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 11 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md:73]: Can find it in the following code-block.

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 12 above)
Comment (Yancey0623) [reply doc/tutorial/didi/carprice_xgboost/README.md:103]: thanks, done.

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 13]
```diff
@@ -1,32 +1,35 @@
...
+how these features affect the car price.
 
-## The Dataset
+## The Car Price Dataset
 
-This tutorial would use the [Car Features and MSRP](https://www.kaggle.com/CooperUnion/cardataset) as the demonstration dataset from kaggle.
+We are using the [Cars Dataset](https://www.kaggle.com/CooperUnion/cardataset) as the demonstration dataset from [kaggle](https://www.kaggle.com/).
 
-We have used feature engineering to preprocess the raw data and create new features. The feature engineering file is `carprice_preprocessing.py` in the current directory. Finally, We separated the dataset into the train/test dataset and got the `trainSet.csv` and `testSet.csv` files, which is used to train/predict our model. 
+We use feature engineering to preprocess the raw data and creating new features, The feature engineering script is `titanic_preprocessing.py` in the current directory.
```
Comment (typhoonzero) [inline doc/tutorial/didi/carprice_xgboost/README.md]: titanic_preprocessing -> carprice_preprocessing

File: doc/tutorial/didi/carprice_xgboost/README.md (hunk 6 above)
Comment (tonyyang-svail) [reply doc/tutorial/didi/carprice_xgboost/README.md]: Hi @typhoonzero @Yancey1989 , I changed it to eighty-two since the number in English is more formal.

File: doc/tutorial/didi/carprice_xgboost/README.md [hunk 14]
```diff
@@ -1,32 +1,35 @@
-# XGBoost on SQLFlow Tutorial
+# Predict the Car Price using XGBoost in SQLFlow
+
+This tutorial describes how to train an XGBoost model using the [Cars Dataset](https://www.kaggle.com/CooperUnion/cardataset), then how to predict the car price using SQLFlow. You can checkout the [User Guide](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/user_guide.md) if you are new to SQLFlow. In this tutorial, you will learn how to:
```
Comment (tonyyang-svail) [inline doc/tutorial/didi/carprice_xgboost/README.md]: **check out** vs **checkout**

> Check out is a verb phrase that means to sign for something or to observe something. Checkout means a place to buy things or the process of leaving a hotel as a noun, and it describes qualities of these things as an adjective.

So in our documentation, we should use check out. :)

**Review by tonyyang-svail — APPROVED** [review]
@Yancey1989 @Echo9573 LGTM++. Great job guys!
